src/main/java/com/epam/tc/api/hw3/service/CommonService.java

```diff
@@ -0,0 +1,76 @@
+package com.epam.tc.api.hw3.service;
+
+import static io.restassured.RestAssured.given;
+
+import com.epam.tc.api.hw3.utils.EndPoints;
+import com.epam.tc.api.hw3.utils.PropertiesReader;
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import java.util.Map;
+
+public class CommonService {
+
+    public static final String KEY = PropertiesReader.getProperty("key");
+    public static final String TOKEN = PropertiesReader.getProperty("token");
+    private RequestSpecification requestSpecification;
+
+    public CommonService() {
+
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+        requestSpecification = new RequestSpecBuilder()
+            .setBaseUri(EndPoints.BASE_URL)
+            .addQueryParam("key", KEY)
+            .addQueryParam("token", TOKEN)
+            .build();
```

Comment:
It looks like you can extract these lines to some method and call it when you need a new common specification (e.g. in putWithParams method or in every method of this class in order to avoid confusion between specifications in different methods). And there will be no need to create new instances of CommonService (lines 29-32)